mrrc/pkgs/indexing.py

```diff
@@ -0,0 +1,99 @@
+from jinja2 import Template
+import sys
+import subprocess
+from treelib import Tree, Node
+import uuid
+import os
+from typing import List
+
+
+class IndexHTML(object):
+    # object for holding index html file data
+    def __init__(self, title: str, header: str, items: List[str]):
+        self.title = title
+        self.header = header
+        self.items = items
+
+    def generate_index_file_content(self) -> str:
+        template = Template(get_index_template())
+        return template.render(index=self)
+
+
+def get_index_template() -> str:
+    DEFAULT_INDEX_TEMPLATE = os.path.join(
+        os.environ["HOME"], ".mrrc/template/index.html.j2"
+    )
+    with open(DEFAULT_INDEX_TEMPLATE, encoding="utf-8") as file_:
+        return file_.read()
+
+
+def generate_from_local(repo):
+    print("Scanning project ...")
+    mavenProcess = subprocess.run(['mvn', '-q', '-Dexec.executable=echo',
```

Comment:
Why should we have a mvn subprocess here? If it is a npm case, this is no sense.
Actually I don't think any package type should be covered in this indexing requirements, what you should consider is only about the file structure listing in directories.